twingraph/src/main/kotlin/com/cosmotech/twingraph/api/TwingraphServiceImpl.kt

```diff
@@ -240,4 +221,96 @@ class TwingraphServiceImpl(
     httpServletResponse!!.setHeader(HttpHeaders.CONTENT_DISPOSITION, contentDisposition.toString())
     return ByteArrayResource(csmJedisPool.resource.use { jedis -> jedis.get(bulkQueryId) })
   }
+
+  private fun getVersions(graphId: String): List<String> {
+    val matchingKeys = mutableSetOf<String>()
+    csmJedisPool.resource.use { jedis ->
+      var nextCursor = ScanParams.SCAN_POINTER_START
+      do {
+        val scanResult = jedis.scan(nextCursor, ScanParams().match("${graphId}:*"), "graphdata")
+        nextCursor = scanResult.cursor
+        matchingKeys.addAll(scanResult.result)
+      } while (!nextCursor.equals(ScanParams.SCAN_POINTER_START))
+    }
+    return matchingKeys.toList()
+  }
+
+  override fun createEntities(
+      organizationId: String,
+      graphId: String,
+      type: String,
+      graphProperties: List<GraphProperties>
+  ): List<ResultSet> {
+    return when (type) {
+      "node" ->
```

Comment:
Convert strings "node" and "relationship" to constants and apply it to others methods too: 

```
const val TYPE_NODE = "node"
const val TYPE_RELATIONSHIP = "relationship"
```